crypto/keys/pkcs11.go

```diff
@@ -0,0 +1,55 @@
+// +build pkcs11
+
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/trillian/crypto/keyspb"
+	"github.com/letsencrypt/pkcs11key"
+)
+
+// Pkcs11ConfigProtoHandler returns a ProtoHandler configured to use the specified PKCS#11 modulePath.
+// This ProtoHandler will retrieve keys as specified by PKCS11Config proto messages.
+// It can be passed to SignerFactory.AddHandler().
+func Pkcs11ConfigProtoHandler(modulePath *string) ProtoHandler {
```

Comment:
That's because this func returns a ProtoHandler func, whereas the other funcs like this *are* the ProtoHandlers. This one is a bit different because the PKCS#11 ProtoHandler needs a bit of configuration (modulePath).